### PR TITLE
[MI-3219] Add documentation to run scripts and add github actions

### PR DESCRIPTION
**Pull request by ayusht2810**
1. Added GitHub actions
2. Update makefile command
3. Added documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,44 @@
-# mattermost-load-test-scripts
+# Mattermost load-test-scripts
+
+Mattermost load-test-scripts provides a set of scripts written in [Go](https://golang.org/) to help profiling [Mattermost](https://github.com/mattermost/mattermost-server) under heavy load, simulating real-world usage of a server installation at scale.
+
+## How to use
+
+- Install K6 load testing tool from [here](https://k6.io/docs/get-started/installation).
```
Comment (raghavaggarwal2308) [inline README.md]:
```suggestion
- Install the K6 load testing tool from [here](https://k6.io/docs/get-started/installation).
```

File: README.md [hunk 2]
```diff
@@ -1 +1,44 @@
...
+
+- Clone the repo using the command
```
Comment (raghavaggarwal2308) [inline README.md]:
```suggestion
- Clone the repo using the command:
```

File: README.md [hunk 3]
```diff
@@ -1 +1,44 @@
...
+- Clone the repo using the command 
+    ```
+    git clone git@github.com:Brightscout/mattermost-load-test-scripts.git
+    ``` 
+    or 
+    ```
+    git clone https://github.com/Brightscout/mattermost-load-test-scripts.git
+    ```
+
+- Download the latest build from the release [here](https://github.com/Brightscout/mattermost-load-test-scripts/releases).
```
Comment (raghavaggarwal2308) [inline README.md]:
```suggestion
- Download the latest build from the releases page [here](https://github.com/Brightscout/mattermost-load-test-scripts/releases).
```

File: README.md [hunk 4]
```diff
@@ -1 +1,44 @@
...
+    - Configure the `config.json` file created according to the load to be tested.
+    - Go to config [docs](docs/config.md) to check details on the config settings.
+
+- Run command `make create_users` to create new users with the details present in the config file.
+
+- Connect the system admin account with the MS Teams account to link the new channels with the MS Teams channel.
+
+- Run command `make create_channels` to create the new channels and add the new users to them. Running this command also links the Mattermost channels with the MS Teams channels present in the config file.
+
+- Run command `make create_dm_and_gm` to create the DM and GM between the new users.
```
Comment (raghavaggarwal2308) [inline README.md]:
```suggestion
- Run command `make create_dm_and_gm` to create the DMs and GMs between the new users.
```

File: docs/config.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+The count of virtual users running concurrently and creating posts in the Mattermost channels, DMs, and GMs.
+
+### MaxSentenceLength
+
+*int*
+
+The maximum length of a sentence in a post.
+
+### MaxWordsLength
+
```
Comment (raghavaggarwal2308) [inline docs/config.md]:
```suggestion
### MaxWordsCount

*int*

The maximum number of words in a sentence in a post.

### MaxWordLength

```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: docs/config.md [hunk 6]
```diff
@@ -86,11 +86,11 @@ The MS Teams Channel ID to which the Mattermost channel is to be linked.
...
 
-### MaxSentenceLength
+### MaxWordsCount
 
 *int*
 
-The maximum length of a sentence in a post.
+The maximum number of words in a sentence in a post.
 
 ### MaxWordsLength
```
Comment (raghavaggarwal2308) [inline docs/config.md]:
```suggestion
### MaxWordLength
```

**Review by raghavaggarwal2308 — APPROVED** [review]

File: .github/ci.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: check style
+        run: make check-style
+        env:
+          TERM: dumb
+
+      - name: Build and bundle the plugin
```
Comment (manojmalik20) [inline .github/ci.yml]: This is not a plugin.

File: .github/ci.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        id: build
+        run: |
+          make build
+          echo ::set-output name=BUNDLE_NAME::$(cd dist; ls *.tar.gz)
+        env:
+          TERM: dumb
+
+      - name: Upload the built plugin as an artifact
```
Comment (manojmalik20) [inline .github/ci.yml]: Same. Fix everywhere.

File: .github/release.yml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Upload Release Assets
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: dist/${{ steps.build.outputs.BUNDLE_NAME }}
+          asset_name: ${{ steps.build.outputs.BUNDLE_NAME }}
+          asset_content_type: application/gzip
```
Comment (manojmalik20) [inline .github/release.yml]: Need for this?

File: README.md [hunk 10]
```diff
@@ -1 +1,44 @@
-# mattermost-load-test-scripts
+# Mattermost load-test-scripts
```
Comment (manojmalik20) [inline README.md]:
```suggestion
# Mattermost load test scripts
```

File: README.md [hunk 11]
```diff
@@ -1 +1,44 @@
...
+    git clone git@github.com:Brightscout/mattermost-load-test-scripts.git
+    ``` 
+    or 
+    ```
+    git clone https://github.com/Brightscout/mattermost-load-test-scripts.git
+    ```
+
+- Download the latest build from the release page [here](https://github.com/Brightscout/mattermost-load-test-scripts/releases).
+
+- Create a folder with the name `dist` in the cloned repo and move the downloaded build into it using the command.
```
Comment (manojmalik20) [inline README.md:20]: This should not be here. The `dist` folder should be created by the build command.

File: README.md [hunk 12]
```diff
@@ -1 +1,44 @@
...
+
+- Create a `config.json` file.
+    - Run command to copy the sample `config.sample.json` file.
+    ```
+    cp config/config.sample.json config/config.json
+    ```
+    - Configure the `config.json` file created according to the load to be tested.
+    - Go to config [docs](docs/config.md) to check details on the config settings.
+
+- Run command `make create_users` to create new users with the details present in the config file.
```
Comment (manojmalik20) [inline README.md:34]: Change this to use the binary file downloaded from the release.

File: README.md [hunk 13]
```diff
@@ -1 +1,44 @@
...
+
+- Connect the system admin account with the MS Teams account to link the new channels with the MS Teams channel.
```
Comment (manojmalik20) [inline README.md]:
```suggestion
- Connect the system admin account with an MS Teams account to link the new channels with the MS Teams channels.
```

File: README.md [hunk 14]
```diff
@@ -1 +1,44 @@
...
+
+- Run command `make create_channels` to create the new channels and add the new users to them. Running this command also links the Mattermost channels with the MS Teams channels present in the config file.
+
+- Run command `make create_dm_and_gm` to create the DMs and GMs between the new users.
+
+- Login and connect all the new users with their respective MS Teams accounts to enable the sharing of messages from Mattermost to MS Teams on behalf of these users.
```
Comment (manojmalik20) [inline README.md]:
```suggestion
- Login and connect all the new users with their respective MS Teams accounts to enable the relaying of messages from Mattermost to MS Teams on behalf of these users.
```

File: docs/config.md [hunk 15]
```diff
@@ -0,0 +1,105 @@
+# Loadtest configurations
```
Comment (manojmalik20) [inline docs/config.md]:
```suggestion
# Load testing configuration
```

File: docs/config.md [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+
+## Connection configurations
```
Comment (manojmalik20) [inline docs/config.md]:
```suggestion
## Connection configuration
```

File: docs/config.md [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+
+### ServerURL
+
+*string*
+
+The URL to direct the load. It should be the public facing URL of the target Mattermost instance.
```
Comment (manojmalik20) [inline docs/config.md]:
```suggestion
### ServerURL: *string*
The URL to direct the load. It should be the public-facing URL of the target Mattermost instance.
```

File: docs/config.md [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+*int*
+
+The maximum number of words in a sentence in a post.
+
+### MaxWordsLength
+
+*int*
+
+The maximum length of each word in a sentence of a post.
```
Comment (manojmalik20) [inline docs/config.md]:
```suggestion
The maximum length of each word in a post message.
```

File: docs/config.md [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+### Duration
+
+*string*
+
+The duration in seconds specifying the total duration of the test run.
```
Comment (manojmalik20) [inline docs/config.md]:
```suggestion
The duration(in seconds) specifying the total duration of the test run.
```

File: .github/workflows/ci.yml [hunk 20]
```diff
@@ -35,15 +35,15 @@ jobs:
...
         id: build
         run: |
           make build
-          echo ::set-output name=BUNDLE_NAME::$(cd dist; ls *.tar.gz)
+          echo ::set-output name=BUNDLE_NAME::$(cd dist; ls)
         env:
           TERM: dumb
 
-      - name: Upload the built plugin as an artifact
+      - name: Upload the built bundle as an artifact
```
Comment (manojmalik20) [inline .github/workflows/ci.yml]:
```suggestion
      - name: Upload the built binary as an artifact
```

File: .github/workflows/ci.yml [hunk 21]
```diff
@@ -35,15 +35,15 @@ jobs:
         env:
           TERM: dumb
 
-      - name: Build and bundle the plugin
+      - name: Build and bundle the scripts
```
Comment (manojmalik20) [inline .github/workflows/ci.yml]:
```suggestion
      - name: Build a binary for the load-tester
```

File: .github/workflows/release.yml [hunk 22]
```diff
@@ -25,11 +25,11 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-go-
 
-      - name: Build and bundle the plugin
+      - name: Build and bundle the scripts
```
Comment (manojmalik20) [inline .github/workflows/release.yml]: Same as previous.

File: .github/workflows/release.yml [hunk 23]
```diff
@@ -25,11 +25,11 @@ jobs:
...
         id: build
         run: |
           make build
-          echo ::set-output name=BUNDLE_NAME::$(cd dist; ls *.tar.gz)
+          echo ::set-output name=BUNDLE_NAME::$(cd dist; ls)
```
Comment (manojmalik20) [inline .github/workflows/release.yml]:
```suggestion
          echo ::set-output name=BINARY_NAME::$(cd dist; ls)
```

File: .github/workflows/release.yml [hunk 24]
```diff
@@ -52,4 +52,4 @@ jobs:
           upload_url: ${{ steps.create_release.outputs.upload_url }}
           asset_path: dist/${{ steps.build.outputs.BUNDLE_NAME }}
           asset_name: ${{ steps.build.outputs.BUNDLE_NAME }}
```
Comment (manojmalik20) [inline .github/workflows/release.yml]:
```suggestion
          asset_path: dist/${{ steps.build.outputs.BINARY_NAME }}
          asset_name: ${{ steps.build.outputs.BINARY_NAME }}
```

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**